app/helpers/quadicon_helper.rb

```diff
@@ -0,0 +1,115 @@
+module QuadiconHelper
+  # render a quadicon
+  #   size: size of the quadicon
+  #   item: element to be represented
+  #   typ:  type of view (:listnav, ....?) # FIXME: find possible values
+  #   db:   name of the model of item
+  #   row:
+  #   height:
+  #   mode:
+  def render_quadicon(item, options)
+    return unless item
+
+    options.reverse_merge! :size => 72
+    options[:db] ||= db_from_item(item)
+
+    if options[:mode] == :text # Rendering the text link, not the quadicon
+      return render(
+               :partial => "layouts/quadicon/quadicon_text",
+               :locals  => {
+                 :db              => options[:db],
+                 :truncate_length => 13,
+                 :row             => options[:row],
+                 :item            => item
+               }
+      )
+    end
+
+    if options[:typ] == :listnav
+      id = ""
+      options[:height] ||= 80
+      style = "margin-left: auto; margin-right: auto; width: 75px; height: #{options[:height]}px; z-index: 0;"
+    else
+      style = ""
+      id = "quadicon"
+    end
+
+    content_tag(:div, :style => style, :id => id) do
+      partial_name = partial_name_from_item(item)
+      # List of removed partials with two lines
+      norender = %w(cim_base_storage_extent cim_storage_extent ontap_file_share ontap_logical_disk ontap_storage_system ontap_storage_volume snia_local_file_system)
+
+      if norender.include?(partial_name)
+        content_tag(:div, :class => 'flobj') do
+          tag(:img, :src => ActionController::Base.helpers.image_path("#{options[:size]}/#{partial_name}.png"), :border => 0)
+        end
+      else
+        render(
+          :partial => "layouts/quadicon/#{partial_name}",
+          :locals  => {
+            :row   => options[:row],
+            :mode  => options[:mode],
+            :size  => options[:size],
+            :width => options[:size] == 150 ? 54 : 35,
+            :item  => item,
+            :typ   => options[:typ],
+            :db    => options[:db]
+          }
+        )
+      end
+    end
+  end
+
+  def db_from_item(item)
+    item.kind_of?(ExtManagementSystem) ? item.class.db_name : item.class.base_model.name
+  end
+
+  def partial_name_from_item(item)
+    partial_name = if %w(EmsCluster ResourcePool Repository Service ServiceTemplate Storage).include?(item.class.name)
+                     item.class.name.underscore
+                   elsif item.kind_of?(VmOrTemplate)
+                     item.class.base_model.to_s.underscore
+                   elsif item.kind_of?(ManageIQ::Providers::Foreman::ConfigurationManager) || item.kind_of?(ManageIQ::Providers::AnsibleTower::ConfigurationManager)
```

Comment:
@Fryguy: one thing after another ;-) Surely I am doing this as a step towards provider plugability. 
But this is out of scope for the current PR.

This is step #1 converting partials into helpers.